webdriver/tests/element_click/element_click.py

```diff
@@ -0,0 +1,111 @@
+
+    # Scroll to the bottom right of the page
+    session.execute_script("window.scrollTo(document.body.scrollWidth, document.body.scrollHeight);")
+    element = session.find.css("input", all=False)
+    # Clear and scroll back to the top of the page
+    response = click(session, element)
+    assert_success(response)
+
+    # Check if element cleared is scrolled into view
+    response = session.execute_script("""rect = document.getElementsByTagName("input")[0].getBoundingClientRect();
+                                        xOffset = window.pageXOffset;
+                                        yOffset = window.pageYOffset;
+                                        return  rect["top"] < (window.innerHeight + yOffset) &&
+                                                rect["left"] < (window.innerWidth + xOffset) &&
+                                                (rect["top"] + rect["height"]) > yOffset &&
+                                                (rect["left"] + rect["width"]) > xOffset;""")
+    assert response is True
+
+
+@pytest.mark.parametrize("transform", ["translate(-50px, -50px)", "rotate(50deg)"])
+
+def test_element_not_interactable(session, transform):
+    # 14.1 Step 5
+    session.url = inline("<div style=\"width: 500px; height: 100px; background-color: green; transform: " + transform + """;">
+                            <a href=#>aaaa</a>
+                        </div>""")
+    element = session.find.css("a", all=False)
+    response = click(session, element)
+    assert_error(response, "element not interactable")
+
+
+@pytest.mark.parametrize("element", ["""<input type=button value=Roger style="position: absolute; left: 10px; top: 10px">
+                                    <div style="position: absolute; height: 100px; width: 100px; background: rgba(255,0,0,.5); left: 10px; top: 5px"></div>""",
+                                    "<input type=button value=Roger style=\"pointer-events: none\">",
+                                    """<div style="position: absolute; height: 50px; width: 100px; background: rgba(255,0,0,.5); left: 10px; top: 50px; overflow: hidden">
+                                        ABCDEFGHIJKLMNOPQRSTUVWXYZ
+                                        <input type=text value=Federer style="position: absolute; top: 50px; left: 10px;">
+                                    </div>"""])
+
+def test_element_click_intercepted(session, element):
+    # 14.1 Step 6
+    # Test intercepted click
+    # Test no pointer events
+    # Test overflow hidden
+    session.url = inline(element)
+
+    element = session.find.css("input", all=False)
+    response = click(session, element)
+    assert_error(response, "element not interactable")
+
+
+def test_match_option_element(session):
+    # 14.1 Step 7
+    session.url = inline("""
+                        <select id=sel style="position: absolute; left: 10px; top: 50px">
```

Comment:
What is the intention of this styling?

Reply:
I wanted the next line to be same indent as the line above it, but I fixed the indentation.